File: exercism/java/hamming/src/main/java/Hamming.java
import java.util.stream.IntStream;

class Hamming {
    private int hammingDistance = 0;

    Hamming(String leftStrand, String rightStrand) {
        if (leftStrand.isEmpty() && !rightStrand.isEmpty()) {
            throw new IllegalArgumentException("left strand must not be empty.");
        } else if (rightStrand.isEmpty() && !leftStrand.isEmpty()) {
            throw new IllegalArgumentException("right strand must not be empty.");
        } else if (leftStrand.length() != rightStrand.length()) {
            throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");
        }
        hammingDistance = (int) IntStream.range(0, leftStrand.length())
                .filter(i -> leftStrand.charAt(i) != rightStrand.charAt(i)).count();
    }

    int getHammingDistance() {
        return hammingDistance;
    }

}

File: exercism/lua/hamming/hamming.lua
local Hamming = {}

function Hamming.compute(a, b)
	if #a ~= #b then
		return -1
	end

	local hamming_distance = 0
	local size = #a
	for i = 1, size do
		if a:sub(i, i) ~= b:sub(i, i) then
			hamming_distance = hamming_distance + 1
		end
	end
	return hamming_distance
end

return Hamming
